src/JLD2.jl

```diff
@@ -334,23 +344,38 @@ function jldopen(fname::AbstractString, wr::Bool, create::Bool, truncate::Bool,
                         throw(ArgumentError("attempted to open file with mmaparrays=$(mmaparrays), but file was already open with mmaparrays=$(f.mmaparrays)"))
                     end
 
+                    f = f::JLDFile{iotype}
+                    f.n_times_opened += 1
+                    return f
+                end
+            elseif parallel_read && haskey(OPEN_FILES, rname)
+                throw(ArgumentError("Tried to open file in a parallel context but it is open elsewhere in a serial context."))
+            elseif parallel_read && haskey(OPEN_PARALLEL_FILES, rname)
+                ref = OPEN_PARALLEL_FILES[rname]
+                f = ref.value
+                if !isnothing(f)
                     f = f::JLDFile{iotype}
                     f.n_times_opened += 1
                     return f
```

Comment:
Sadly, I think, this does not solve the problem at all.

With this logic you still just end up with a single open file handle.
This should should return a new file handle each time.